linera-explorer-new/src/types/blockchain.ts

```diff
@@ -44,13 +44,81 @@ export interface PostedMessage {
   grant_amount: number;
   refund_grant_to: Uint8Array | null;
   message_kind: string;
+  message_type?: 'System' | 'User';
+  application_id?: string;
+  system_message_type?: string;
+  system_target?: string;
+  system_amount?: number;
+  system_source?: string;
+  system_owner?: string;
+  system_recipient?: string;
   message_data: Uint8Array;
   created_at: string;
 }
 
+export interface IncomingBundleWithMessages extends IncomingBundle {
+  messages: PostedMessage[];
+}
```

Comment:
This should be called `IncomingBundle` since having incoming messages is the default.